src/main/kotlin/com/openlattice/transporter/services/DataTransporterService.kt

```diff
@@ -0,0 +1,129 @@
+        val pollTimer: Histogram = Histogram.build()
+                .namespace("transporter")
+                .name("poll_duration_seconds")
+                .help("time to do one database sync poll")
+                .register()
+    }
+
+    private lateinit var acm: AssemblerConnectionManager
+    private lateinit var transporter: HikariDataSource
+
+    private val transporterState = HazelcastMap.TRANSPORTER_DB_COLUMNS.getMap( hazelcastInstance )
+
+    private fun syncTable(et: EntityType): Optional<Future<*>> {
+        val prev = transporterState.putIfAbsent(et.id, TransporterColumnSet(emptyMap()))
+        return if (prev == null) {
+            val props = dataModelService.getPropertyTypes(et.properties)
+            Optional.of(
+                    transporterState.submitToKey(et.id, TransporterSynchronizeTableDefinitionEntryProcessor(props).init(acm))
+            )
+        } else {
+            Optional.empty()
+        }
+    }
+
+    override fun init(acm: AssemblerConnectionManager): Void? {
+        this.acm = acm
+        this.transporter = acm.connect("transporter")
+        executor.submit {
+            val entityTypes = dataModelService.entityTypes.toList()
+            logger.info("initializing DataTransporterService with {} types", entityTypes.size)
+            val tablesCreated = entityTypes.map { et -> this.syncTable(et) }
+                    .filter { it.isPresent }
+                    .map { it.get().get() }
+                    .count()
+            logger.info("synchronization finished with {} entity type tables updated", tablesCreated)
+        }
+
+        eventBus.register(this)
+        return null
+    }
+
+    private fun partitions(entitySetIds: Set<UUID>): Collection<Int> {
+        return partitionManager.getPartitionsByEntitySetId(entitySetIds).values.flatten().toSet()
+    }
+
+    fun pollOnce() {
+        val timer = pollTimer.startTimer()
+        val futures = this.transporterState.keys.map { entityTypeId ->
+            val relevantEntitySets = validEntitySets(entityTypeId)
+            val entitySetIds = relevantEntitySets.map { it.id }.toSet()
+            val partitions = partitions(entitySetIds)
+            val ft = transporterState.submitToKey(entityTypeId, TransporterPropagateDataEntryProcessor(relevantEntitySets, partitions).init(acm))
+            entitySetIds.count() to ft
+        }
+        futures.forEach { it.second.get() }
+        val setsPolled = futures.map { it.first }.sum()
+        val duration = timer.observeDuration()
+        logger.debug("Total poll duration time for {} entity sets in {} entity types: {} sec", setsPolled, futures.size, duration)
+    }
+
+    private fun validEntitySets(entityTypeId: UUID): Set<EntitySet> {
+        return entitySetService.getEntitySetsOfType(entityTypeId)
+                .filter { !it.isLinking && !it.flags.contains(EntitySetFlag.AUDIT) }.toSet()
```

Comment:
How/where will be linking entity sets be materialized?

Reply:
views for linking entity sets shouldn't need to be copied explicitly.

```
create or replace view "EntitySetName" as
select
  es.id as entity_set_id,
  ids.linking_id as id, -- todo: obfuscate
  array_agg("property-uuid") FILTER (WHERE "property-uuid" is not null) as "property.name", -- multivalued = true
  (array_agg("property-uuid") FILTER (WHERE "property-uuid" is not null))[1] as "property.name", -- multivalued = false
from entity_sets es 
  join ids on (ids.entity_set_id = any(es.linked_entity_sets))
  join "et_entity-type-uuid" t on (ids.id = t.id)
where es.id = 'entity-set-uuid'
  and es.linking_id is not null
group by es.id, ids.linking_id;
```

(We will also need to create views for non-linked entity sets to deal with subsetting by entity set id, using property name instead of uuid for columns, etc).